torchmetrics/metric.py

```diff
@@ -377,7 +378,19 @@ def wrapped_func(*args: Any, **kwargs: Any) -> None:
             self._computed = None
             self._update_count += 1
             with torch.set_grad_enabled(self._enable_grad):
-                update(*args, **kwargs)
+                try:
+                    update(*args, **kwargs)
+                except RuntimeError as e:
+                    if "Expected all tensors to be on" in str(e):
+                        traceback.print_exc()
```

Comment:
is this needed while you raise the error below?

Reply:
To still get the original traceback. It could be that the error was raised for some other reason than the metric was on the wrong device and in that case I think it is great to still have the original traceback.

Reply:
In that case, you should be using the `raise RuntimeError(...) from err` construct which will do what you intend and then you don't need to print the previous exception manually. One example in PL:

https://github.com/PyTorchLightning/pytorch-lightning/blob/7938293cd9bb27e5ad97aabc7261fb29027dc6b3/pytorch_lightning/utilities/data.py#L201

Reply:
cool, will do :]